pkg/controllers/clusterset/syncrolebinding/syncrolebinding_controller.go

```diff
@@ -0,0 +1,120 @@
+		clustersetToClusters:  clustersetToClusters,
+		clustersetToNamespace: clustersetToNamespace,
+	}
+}
+
+// start a routine to sync the clusterrolebinding periodically.
+func (r *Reconciler) Run(period time.Duration) {
+	go utilwait.Forever(r.reconcile, period)
+}
+
+//This function sycn the rolebinding in namespace which in r.clustersetToNamespace and r.clustersetToClusters
+func (r *Reconciler) reconcile() {
+	ctx := context.Background()
+
+	//union the clusterset to namespace and clusterset to cluster(it's same as managedcluster namespace).
+	//so we can use unionclustersetToNamespace to generate role bindings.
+	unionclustersetToNamespace := r.clustersetToNamespace.UnionObjectsInClusterSet(r.clustersetToClusters)
+	clustersetToAdminSubjects := clustersetutils.GenerateClustersetSubjects(r.clusterSetAdminCache)
+	clustersetToViewSubjects := clustersetutils.GenerateClustersetSubjects(r.clusterSetViewCache)
+
+	r.syncRoleBinding(ctx, unionclustersetToNamespace, clustersetToAdminSubjects, "admin")
+	r.syncRoleBinding(ctx, unionclustersetToNamespace, clustersetToViewSubjects, "view")
+
+}
+
+func (r *Reconciler) syncRoleBinding(ctx context.Context, clustersetToNamespace *helpers.ClusterSetMapper, clustersetToSubject map[string][]rbacv1.Subject, role string) []error {
+	namespaceToSubject := clustersetutils.GenerateObjectSubjectMap(clustersetToNamespace, clustersetToSubject)
+	//apply all disired clusterrolebinding
+	errs := []error{}
+	for namespace, subjects := range namespaceToSubject {
+		requiredRoleBinding := generateRequiredRoleBinding(namespace, subjects, role)
+		err := utils.ApplyRoleBinding(ctx, r.kubeClient, requiredRoleBinding)
+		if err != nil {
+			klog.Errorf("Failed to apply rolebinding: %v, error:%v", requiredRoleBinding.Name, err)
+			errs = append(errs, err)
+		}
+	}
+
+	//Delete rolebinding
+	roleBindingList, err := r.kubeClient.RbacV1().RoleBindings("").List(ctx, metav1.ListOptions{LabelSelector: clustersetutils.ClusterSetLabel})
+	if err != nil {
+		klog.Errorf("Error to list clusterrolebinding. error:%v", err)
+	}
+	for _, roleBinding := range roleBindingList.Items {
+		curRoleBinding := roleBinding
+
+		//only handle current resource rolebinding
+		matchRoleBindingName := utils.GenerateClustersetResourceRoleBindingName(role)
+
+		if matchRoleBindingName != curRoleBinding.Name {
+			continue
+		}
+
+		if _, ok := namespaceToSubject[roleBinding.Namespace]; !ok {
+			err := r.kubeClient.RbacV1().RoleBindings(curRoleBinding.Namespace).Delete(ctx, curRoleBinding.Name, metav1.DeleteOptions{})
+			if err != nil {
+				klog.Errorf("Error to delete clusterrolebinding, error:%v", err)
+			}
+		}
+	}
+	return errs
+}
+
+func generateRequiredRoleBinding(resourceNameSpace string, subjects []rbacv1.Subject, role string) *rbacv1.RoleBinding {
+	roleBindingName := utils.GenerateClustersetResourceRoleBindingName(role)
+
+	var labels = make(map[string]string)
+	labels[clustersetutils.ClusterSetLabel] = "true"
```

Comment:
what is the intention of this label?

Reply:
it's used to filter rolebinding when we did the sync logic. with this label, we can only list clusterset related rolebinding.

Reply:
can you set labels[clustersetutils.ClusterSetLabel] = clusterset and use match expression to filter?

Reply:
done